### PR TITLE
update documentations

### PR DESCRIPTION
**Pull request by agirodi**
This PR adds:

- changes of mentions from "--no_cuda" to "--device cpu"
- update mentions from "fsx.x" to "fs7.3.2" (some revision may be needed)
- update of some flags (mostly of eval.py and generate_hdf5.py)

### CODE REVIEW

File: FastSurferCNN/README.md [hunk 1]
```diff
@@ -10,20 +10,25 @@ The network was trained with conformed images (UCHAR, 256x256x256, 1 mm voxels a
 The *FastSurferCNN* directory contains all the source code and modules needed to run the scripts. A list of python libraries used within the code can be found in __requirements.txt__. The main script is called __eval.py__ within which certain options can be selected and set via the command line:
```
Comment (LeHenschel) [inline FastSurferCNN/README.md]: It was decided to call the script run_prediction.py and not eval.py anymore in vinn (more descriptive). So these commands need to be changed in the documentation for vinn

File: FastSurferCNN/README.md [hunk 2]
```diff
@@ -10,20 +10,25 @@ The network was trained with conformed images (UCHAR, 256x256x256, 1 mm voxels a
...
-* --out_name: name of the prediction; the extension determines the output file format, which can be one of: mgz, nii, or nii.gz (__default: aparc.DKTatlas+aseg.deep.mgz__)
-* --log: name of log-file (information about processing is stored here; __default: deep-seg.log__). Saved in the same directory as the predictions.
-* --order: order of interpolation (0=nearest,__1=linear(default)__, 2=quadratic, 3=cubic) for conformation (if not already done).
+* --sd: Path to output directory (where should predictions be saved). Will be created if it does not already exist.
+* --seg_log: name of log-file (information about processing is stored here; If not set, logs will not be saved). Saved in the same directory as the predictions.
+* --strip: strip suffix from path definition of input file to yield correct subject name. (Optional, if full path is defined for --t1)
+* --lut: FreeSurfer-style Color Lookup Table with labels to use in final prediction. Default: ./config/FastSurfer_ColorLUT.tsv
+* --gn: How often to sample from gaussian and run inference on same sample with added noise on scale factor
```
Comment (LeHenschel) [inline FastSurferCNN/README.md]: This should be removed. Not sure why this is still in here. Is this also present in the run_prediction.py script?

File: FastSurferCNN/README.md [hunk 3]
```diff
@@ -33,19 +38,21 @@ The *FastSurferCNN* directory contains all the source code and modules needed to
...
+* --batch_size: Batch size for inference. Default=8
 
 
 ### Example Command Evaluation Single Subject
 To run the network on MRI-volumes of subject1 in ./data (specified by --i_dir flag; e.g. ./data/subject1/orig.mgz), change into the *FastSurferCNN* directory and run the following commands: 
 
 ```
-python3 eval.py --i_dir ../data \
---o_dir ../data \
+python3 eval.py --in_dir ../data \
```
Comment (LeHenschel) [inline FastSurferCNN/README.md]: see previous comment run_prediction.py

File: FastSurferCNN/README.md [hunk 4]
```diff
@@ -33,19 +38,21 @@ The *FastSurferCNN* directory contains all the source code and modules needed to
...
+--sd ../data \
 --t subject1 \
---log temp_Competitive.log \
---network_sagittal_path ../checkpoints/Sagittal_Weights_FastSurferCNN/ckpts/Epoch_30_training_state.pkl \
---network_coronal_path ../checkpoints/Coronal_Weights_FastSurferCNN/ckpts/Epoch_30_training_state.pkl \
---network_axial_path ../checkpoints/Axial_Weights_FastSurferCNN/ckpts/Epoch_30_training_state.pkl 
+--sg_log temp_Competitive.log \
```
Comment (LeHenschel) [inline FastSurferCNN/README.md]: --seg_log 
 
Example should include a full path (or it should be mentioned below that this is possible).

File: FastSurferCNN/README.md [hunk 5]
```diff
@@ -54,12 +61,12 @@ The output will be saved in ./data/subject1/aparc.DKTatlas+aseg.deep.mgz.
...
---network_axial_path ../checkpoints/Axial_Weights_FastSurferCNN/ckpts/Epoch_30_training_state.pkl 
+python3 eval.py --in_dir ../data \
+--sd ../data \
+--seg_log temp_Competitive.log \
+--ckpt_sag ../checkpoints/Sagittal_Weights_FastSurferCNN/ckpts/Epoch_30_training_state.pkl \
+--ckpt_cor ../checkpoints/Coronal_Weights_FastSurferCNN/ckpts/Epoch_30_training_state.pkl \
+--ckpt_ax ../checkpoints/Axial_Weights_FastSurferCNN/ckpts/Epoch_30_training_state.pkl 
 ```
 
 The output will be stored in ./data/subjectX/aparc.DKTatlas+aseg.deep.mgz.
```
Comment (LeHenschel) [inline FastSurferCNN/README.md]: Here, it should be mentioned, that the log-file created here will have all outputs (all subjects logged into one) and that is also possible to leave the log-file undefined in which case everything is written to stdout.

File: FastSurferCNN/README.md [hunk 6]
```diff
@@ -70,8 +77,8 @@ The *FastSurferCNN* directory contains all the source code and modules needed to
 A list of python libraries used within the code can be found in __requirements.txt__. The main script is called __generate_hdf5.py__ within which certain options can be selected and set via the command line:
 
 #### General
-* --hdf5_name: Path and name of the to-be-created hdf5-file
-* --data_dir: Directory with images to load
+* --hdf5_name: Path and name of the to-be-created hdf5-file. Default: ../data/hdf5_set/Multires_coronal.hdf5
+* --data_dir: Directory with images to load. Default: /testsuite
```
Comment (LeHenschel) [inline FastSurferCNN/README.md]: Default should be /data. Should also be changed in run_prediction.py

**Review by LeHenschel — CHANGES_REQUESTED** [review]
See comments in file and add changes. Let me know if anything is unclear

**Comment by LeHenschel** [comment]
README.md:

> with a resolution close to 1mm isotropic

This is not true for VINN. Higher resolutions work as well (tested up to 0.7)

> Sub-mm scans (e.g. .75 or .8mm isotropic) will be downsampled by us automatically to 1mm isotropic

Not true, when the hires flag is set, so this needs to be updated

> 256x256x256, 1 mm voxels

Again, this is not the case for VINN (conformed to minimum voxelsize)

- System requirements should be checked (same for VINN?). David did some tests on this

FastSurferCNN/README.md
- Image needs to be updated (VINN graphic, not CNN graphic)
- Change the passages refering to a.) eval.py and b.) only 1 mm processing (see also comments for README.md)
> The network was trained with conformed images (UCHAR, 256x256x256, 1 mm voxels and standard slice orientation). These specifications are checked in the eval.py script and the image is automatically conformed if it does not comply.

- Some information what multiple resolution means in the example for hdf5-generation
- Height and Width are no longer args in generare_hdf5.py (should be replaced with sizes)
- Typo in MODEL_NAME (MODEL_NAMEL) in training section
- Default for sizes should be 256, 311 and 320
- Default for augmentations should just be scaling and translation

Docker/README.md
- First example should be changed to just pulling docker from our repo and running it
- Name of self-build should be updated to include fastsurferversion
- '--user 123X' should be changed to $(id -u):$(id -g), explanation what the commands mean
- FastSurferCNN docker do not have the correct flags (--t1 instead of --in_name, --seg instead of --out_name, ...)


__Make sure, that the commands actually work after updating__

File: FastSurferCNN/README.md (hunk 2 above)
Comment (agirodi) [reply FastSurferCNN/README.md]: Yes they are. All the variables here also seem to still be in use or am I missing something?

Comment (LeHenschel) [reply FastSurferCNN/README.md]: This one is (or should not) be used in run_prediction.py: --gn: How often to sample from gaussian and run inference on same sample with added noise on scale factor

Comment (agirodi) [reply FastSurferCNN/README.md]: The gn goes as far as being passed from run_prediction.py to inference.py to dataset.py. I'm assuming it should only be deleted int run_predictions.py so it defaults to 0, correct?
[run_prediction.py#L174](https://github.com/Deep-MI/FastSurfer/blob/9c467e27019171ed34ad9daf8384a349473933e4/FastSurferCNN/run_prediction.py#L174) -> [inference.py#L191](https://github.com/Deep-MI/FastSurfer/blob/9c467e27019171ed34ad9daf8384a349473933e4/FastSurferCNN/inference.py#L191) -> [dataset.py#L81](https://github.com/Deep-MI/FastSurfer/blob/9c467e27019171ed34ad9daf8384a349473933e4/FastSurferCNN/data_loader/dataset.py#L81)

**Comment by agirodi** [comment]
Changes done:
* update or add tag "v2.0.0" to docker images
* update docs for 1-0.7mm support
* fix (unused) function data_utils.load_and_conform_image() not supporting <0.7mm images
* update flags (in hdf5.py)
* change "--user X" flag to "--user $(id -u):$(id -g)" (and its docs)
* add example 1 "docker pull"-use case
* update eval.py to run_prediction.py
* update checkpoint names
* add/update some docs

File: FastSurferCNN/README.md (hunk 2 above)
Comment (LeHenschel) [reply FastSurferCNN/README.md]: All of the parts with gn_noise should be removed. All the checks in run_prediction.py, inference.py and dataset.py. This was just for testing an should not be part of the official release

File: Docker/README.md [hunk 7]
```diff
@@ -79,24 +101,22 @@ For running the analysis, start the container (e.g. to run segmentation on __all
 ```bash
 docker run --gpus all -v /home/user/my_mri_data:/data \
                       -v /home/user/my_fastsurferCNN_analysis:/output \
-                      --rm --user 123X fastsurfercnn:gpu \
-                      --i_dir /data \
-                      --in_name mri/orig.mgz \
-                      --o_dir /output \
-                      --out_name mri/aparc.DKTatlas+aseg.deep.mgz \
-                      --log deep_surfer.log
+                      --rm --user $(id -u):$(id -g) fastsurfercnn:gpu \
```
Comment (LeHenschel) [inline Docker/README.md]: Might make sense to keep this consistent, i.e. deepmi/fastsurfer:gpu-segonly-v2.0.0. @m-reuter what do you think?

File: Docker/README.md [hunk 8]
```diff
@@ -106,20 +126,18 @@ docker build --rm=true -t fastsurfercnn:cpu -f ./Docker/Dockerfile_FastSurferCNN
...
-           --rm --user 123X fastsurfercnn:cpu \
-           --i_dir /data \
-           --in_name mri/orig.mgz \
-           --o_dir /output \
-           --out_name mri/aparc.DKTatlas+aseg.deep.mgz \
-           --log deep_surfer.log \
-           --no_cuda
+docker run --gpus all -v /home/user/my_mri_data:/data \
+                      -v /home/user/my_fastsurferCNN_analysis:/output \
+                      --rm --user $(id -u):$(id -g) fastsurfercnn:cpu \
```
Comment (LeHenschel) [inline Docker/README.md]: Might make sense to keep this consistent, i.e. deepmi/fastsurfer:gpu-segonly-v2.0.0. @m-reuter what do you think?

File: Docker/README.md [hunk 9]
```diff
@@ -133,23 +151,23 @@ For running the analysis, start the container (e.g. to run segmentation on __all
 docker run -v /home/user/my_mri_data:/data \
            -v /home/user/my_fastsurfer_analysis:/output \
            -v /home/user/my_fs_license_dir:/fs_license \
-           --rm --user 123X fastsurfer_reconsurf:cpu \
+           --rm --user $(id -u):$(id -g) fastsurfer_reconsurf:cpu \
```
Comment (LeHenschel) [inline Docker/README.md]: Same as above (deepmi/fastsurfer:cpu-surfonly-v2.0.0)

File: FastSurferCNN/README.md [hunk 10]
```diff
@@ -1,29 +1,34 @@
 # Overview
 
 This directory contains all information needed to run inference with the readily trained FastSurferCNN or train it from scratch. FastSurferCNN is capable of whole brain segmentation into 95 classes in under 1 minute, mimicking FreeSurfer's anatomical segmentation and cortical parcellation (DKTatlas). The network architecture incorporates local and global competition via competitive dense blocks and competitive skip pathways, as well as multi-slice information aggregation that specifically tailor network performance towards accurate segmentation of both cortical and sub-cortical structures. 
-![](/images/FastSurfer_v5.png)
+![](/images/detailed_network.png)
 
-The network was trained with conformed images (UCHAR, 256x256x256, 1 mm voxels and standard slice orientation). These specifications are checked in the eval.py script and the image is automatically conformed if it does not comply.
+The network was trained with conformed images (UCHAR, 256x256x256, 1-0.7 mm voxels and standard slice orientation). These specifications are checked in the run_prediction.py script and the image is automatically conformed if it does not comply.
```
Comment (LeHenschel) [inline FastSurferCNN/README.md]: 256x256x256 is not correct. Images get conformed to min size (which is 256 for 1.0 mm)

File: FastSurferCNN/README.md [hunk 11]
```diff
@@ -1,29 +1,34 @@
...
 
-The *FastSurferCNN* directory contains all the source code and modules needed to run the scripts. A list of python libraries used within the code can be found in __requirements.txt__. The main script is called __eval.py__ within which certain options can be selected and set via the command line:
+The *FastSurferCNN* directory contains all the source code and modules needed to run the scripts. A list of python libraries used within the code can be found in __requirements.txt__. The main script is called __run_prediction.py__ within which certain options can be selected and set via the command line:
 
 #### General
-* --i_dir: Path to the input volume directory (e.g /your/path/to/ADNI/fs60) or 
+* --in_dir: Path to the input volume directory (e.g /your/path/to/ADNI/fs60) or 
 * --csv_file: Path to csv-file listing input volume directories
-* --in_name: name of the MRI_volume (like mri_volume.mgz, __default: orig.mgz__)
+* --t1: name of the MRI_volume (like mri_volume.mgz, __default: orig.mgz__)
```
Comment (LeHenschel) [inline FastSurferCNN/README.md]: Add T1-weighted MRI volume

File: FastSurferCNN/generate_hdf5.py [hunk 12]
```diff
@@ -224,7 +224,7 @@ def create_hdf5_dataset(self, plane='axial'):
                         help="Turn on to add cortex mask for hires-processing.")
     parser.add_argument('--processing', type=str, default="aparc", choices=["aparc", "aseg", "none"],
                         help="Use aseg, aparc or no specific mapping processing")
-    parser.add_argument('--sizes', nargs='+', type=int, default=256)
+    parser.add_argument('--sizes', nargs='+', type=int, default=256, help="Resolutions of images in the dataset. Default: 256")
```
Comment (LeHenschel) [inline FastSurferCNN/generate_hdf5.py]: This is not resolution, but image size (which in FreeSurfer depends on the resolution). But here this is just the image size.

File: FastSurferCNN/README.md [hunk 13]
```diff
@@ -237,8 +245,8 @@ The configuration options include:
 
 * BATCH_SIZE: Input batch size for training. Default: 16
 * NUM_EPOCHS: Number of epochs to train. Default: 30
-* SIZES: Available image sizes for the multi-scale dataloader. Default: [128, 183, 257, 256, 311, 320]
-* AUG: Augmentations. Default: ["Flip", "Elastic", "Scaling", "Rotation", "Translation", "RAnisotropy", "BiasField", "RGamma"]
+* SIZES: Available image sizes for the multi-scale dataloader. Default: [256, 311 and 320]
```
Comment (LeHenschel) [inline FastSurferCNN/README.md]: This should also be changed in the default.py script

File: FastSurferCNN/run_prediction.py [hunk 14]
```diff
@@ -304,7 +304,7 @@ def handle_cuda_memory_exception(exception: RuntimeError, exit_on_out_of_memory:
                         help="Directory in which evaluation results should be written. "
                              "Will be created if it does not exist. Optional if full path is defined for --pred_name.")
     parser.add_argument('--hires', action="store_true", default=False, dest='hires',
-                        help="Switch on hires processing (no conforming to 1mm, but to smallest voxel size.")
+                        help="Switch on hires processing (no conforming to 1mm, but to smallest voxel size).")
```
Comment (LeHenschel) [inline FastSurferCNN/run_prediction.py]: Description for hires flag is missing in run_fastsurfer.sh. Should be added there as well.

**Review by LeHenschel — CHANGES_REQUESTED** [review]
Thanks for the updates. Still a few things that should be updated (see individual comments)

**Comment by agirodi** [comment]
Changes done: 

- remove 256x256x256 [FastSurferCNN/README.md#L6](https://github.com/agirodi/FastSurfer/tree/b4b0b978d4176a6e64afa7f6feaccdfd5cd7c0ed/FastSurferCNN/README.md?plain=1#L6)
- add T1-weighted MRI volume [FastSurferCNN/README.md#L15](https://github.com/agirodi/FastSurfer/tree/b4b0b978d4176a6e64afa7f6feaccdfd5cd7c0ed/FastSurferCNN/README.md?plain=1#L15)
- change wording "Resolution" to "Sizes" [FastSurferCNN/generate_hdf5.py#L227](https://github.com/agirodi/FastSurfer/tree/b4b0b978d4176a6e64afa7f6feaccdfd5cd7c0ed/FastSurferCNN/generate_hdf5.py#L227)
- change sizes and augmentations defaults [FastSurferCNN/config/defaults.py#L143](https://github.com/agirodi/FastSurfer/tree/b4b0b978d4176a6e64afa7f6feaccdfd5cd7c0ed/FastSurferCNN/config/defaults.py#L143)
- add description for hires flag [run_fastsurfer.sh#L84](https://github.com/agirodi/FastSurfer/tree/b4b0b978d4176a6e64afa7f6feaccdfd5cd7c0ed/run_fastsurfer.sh#L84)
- change wording "hires" to "high resolution" [FastSurferCNN/README.md?plain=1#L41](https://github.com/agirodi/FastSurfer/tree/b4b0b978d4176a6e64afa7f6feaccdfd5cd7c0ed/FastSurferCNN/README.md?plain=1#L41), [FastSurferCNN/README.md#L109](https://github.com/agirodi/FastSurfer/tree/b4b0b978d4176a6e64afa7f6feaccdfd5cd7c0ed/FastSurferCNN/README.md?plain=1#L109), [FastSurferCNN/run_prediction.py#L307](https://github.com/agirodi/FastSurfer/tree/b4b0b978d4176a6e64afa7f6feaccdfd5cd7c0ed/FastSurferCNN/run_prediction.py#L307)

**Comment by agirodi** [comment]
Changes done: 
- Removing all the parts with [gn_noise](https://github.com/Deep-MI/FastSurfer/pull/206#discussion_r1028285786)

File: FastSurferCNN/data_loader/loader.py [hunk 15]
```diff
@@ -106,8 +106,7 @@ def get_dataloader(cfg, mode):
             all_augs = {"Elastic": elastic, "Scaling": scaling, "Rotation": rot, "Translation": tl,
                         "RAnisotropy": ra, "BiasField": bias_field, "RGamma": random_gamma}
 
-            all_tfs = {all_augs[aug]: 0.8 for aug in cfg.DATA.AUG if aug != "Gaussian"}
-            gaussian_noise = True if "Gaussian" in cfg.DATA.AUG else False
+            all_tfs = {all_augs[aug]: 0.8 for aug in cfg.DATA.AUG}
```
Comment (LeHenschel) [inline FastSurferCNN/data_loader/loader.py]: the if aug != "Gaussian" should be kept

File: FastSurferCNN/data_loader/loader.py [hunk 16]
```diff
@@ -106,8 +106,7 @@ def get_dataloader(cfg, mode):
             all_augs = {"Elastic": elastic, "Scaling": scaling, "Rotation": rot, "Translation": tl,
                         "RAnisotropy": ra, "BiasField": bias_field, "RGamma": random_gamma}
 
-            all_tfs = {all_augs[aug]: 0.8 for aug in cfg.DATA.AUG if aug != "Gaussian"}
-            gaussian_noise = True if "Gaussian" in cfg.DATA.AUG else False
```
Comment (LeHenschel) [inline FastSurferCNN/data_loader/loader.py]: This should not be removed as it is important for the data loader during training! This is different from gn_noise during inferende!

File: FastSurferCNN/data_loader/loader.py [hunk 17]
```diff
@@ -116,7 +115,7 @@ def get_dataloader(cfg, mode):
 
             logger.info(f"Loading {mode.capitalize()} data ... from {data_path}. Using torchio Aug")
 
-            dataset = dset.MultiScaleDataset(data_path, cfg, gaussian_noise, transform)
+            dataset = dset.MultiScaleDataset(data_path, cfg, transform)
```
Comment (LeHenschel) [inline FastSurferCNN/data_loader/loader.py]: Again, during training gaussian noise is needed, not during inference only. So gaussian_noise should be added here again

File: Docker/README.md (hunk 8 above)
Comment (LeHenschel) [reply Docker/README.md]: This should be updated similar to the changes above (line 81)

File: Docker/README.md (hunk 7 above)
Comment (LeHenschel) [reply Docker/README.md]: Should be updated akin to line 81

File: Docker/README.md (hunk 9 above)
Comment (LeHenschel) [reply Docker/README.md]: update here as well

File: FastSurferCNN/data_loader/dataset.py [hunk 18]
```diff
@@ -101,11 +97,10 @@ class MultiScaleDataset(Dataset):
     """
     Class for loading aseg file with augmentations (transforms)
     """
-    def __init__(self, dataset_path, cfg, gn_noise=False, transforms=None):
```
Comment (LeHenschel) [inline FastSurferCNN/data_loader/dataset.py]: Please revert - during training (in MultiScaleDataset and MultiScaleDatasetVal) gaussian noise is wanted and needed! Only during inference the lines are irrelevant/unneeded!

File: FastSurferCNN/data_loader/dataset.py [hunk 19]
```diff
@@ -101,11 +97,10 @@ class MultiScaleDataset(Dataset):
...
+    def __init__(self, dataset_path, cfg, transforms=None):
 
         self.max_size = cfg.DATA.PADDED_SIZE
         self.base_res = cfg.MODEL.BASE_RES
-        self.gn_noise = gn_noise
```
Comment (LeHenschel) [inline FastSurferCNN/data_loader/dataset.py]: See above

File: FastSurferCNN/data_loader/dataset.py [hunk 20]
```diff
@@ -162,10 +157,6 @@ def _get_scale_factor(self, img_zoom, scale_aug):
 
         scale = self.base_res / img_zoom
 
-        if self.gn_noise:
```
Comment (LeHenschel) [inline FastSurferCNN/data_loader/dataset.py]: See above

**Review by LeHenschel — CHANGES_REQUESTED** [review]
Thanks for the changes. The gaussian_noise flag should only be removed for inference/running the prediction. During training this flag is important and still needed. So for the classes in dataset.py, this should only be removed from MultiScaleOrigDataThisSlices, not MultiScaleDataset. Similarly, it should not be removed from loader.py at all (only used for training).

**Comment by agirodi** [comment]
Changes done: 
- consistent tags in [Docker/README.md](https://github.com/agirodi/FastSurfer/blob/b4b0b978d4176a6e64afa7f6feaccdfd5cd7c0ed/Docker/README.md?plain=1#L104)
- revert some previous changes to gaussian norm in 1cf991cfdb759ca7d43cfd44d76399b8f168f7b5

**Review by LeHenschel — APPROVED** [review]

**Comment by LeHenschel** [comment]
Update now includes all relevant changes from what I can tell. The branch conflicts need to be solved and afterwards the branch can be merged.

**Comment by dkuegler** [comment]
Retargetting to dev

**Pull request merged**